app/src/adapters/db/flask_db.py

```diff
@@ -73,33 +80,42 @@ def get_db(app: Flask) -> DBClient:
         def health():
             db_client = flask_db.get_db(current_app)
     """
-    return app.extensions[_FLASK_EXTENSION_KEY]
+    return app.extensions[client_name]
 
 
 P = ParamSpec("P")
 T = TypeVar("T")
 
 
-def with_db_session(f: Callable[Concatenate[db.Session, P], T]) -> Callable[P, T]:
+def with_db_session(
+    *, client_name: str = _DEFAULT_FLASK_EXTENSION_KEY
+) -> Callable[[Callable[Concatenate[db.Session, P], T]], Callable[P, T]]:
     """Decorator for functions that need a database session.
 
     This decorator will create a new session object and pass it to the function
     as the first positional argument. A transaction is not started automatically.
     To start a transaction use db_session.begin()
 
     Usage:
-        @with_db_session
+        @with_db_session()
         def foo(db_session: db.Session):
             ...
 
-        @with_db_session
+        @with_db_session()
         def bar(db_session: db.Session, x, y):
             ...
+
+        @with_db_session(client_name="not_the_primary_client")
+        def fiz(db_session: db.Session, x, y, z):
+            ...
     """
 
-    @wraps(f)
-    def wrapper(*args: Any, **kwargs: Any) -> T:
-        with get_db(current_app).get_session() as session:
-            return f(session, *args, **kwargs)
+    def decorator_func(f: Callable[Concatenate[db.Session, P], T]) -> Callable[P, T]:
```

Comment:
nit, i thought a decorator is always a function, so maybe just call it `decorator` instead of `decorator_func`
(btw don't feel too strongly about this comment)